packages/common/test/util.js

```diff
@@ -1370,4 +1370,30 @@ describe('common/util', function() {
       });
     });
   });
+
+  describe('isCustomType', function() {
+    function PubSub() {}
+
+    function MiddleLayer() {
+      this.parent = new PubSub();
+    }
+
+    function Subscription() {
+      this.parent = new MiddleLayer();
+    }
+
+    var subscription = new Subscription();
+
+    it('should match a custom type by constructor names', function() {
```

Comment:
Not sure if it will ever come up, but what do you think about adding a test to check the type of a `new PubSub()` - testing against Service and not a ServiceObject.